### PR TITLE
Documentation rewrite

### PR DESCRIPTION
**Pull request by ColCarroll**
I think this is ready to go. This dramatically reorganizes the documentation. A live demo is being hosted at
 https://colindcarroll.com/static/pymc3/

Note that some of the links are broken in the demo because absolute links will start my website's root, not the documentation root (they will work on the actual doc site).

Major changes:
- UI and CSS is custom, using semantic-ui. I did this because it looks clean and different from bootstrap, so it feels unique. most of these changes are in `layout.html`
- Example gallery has pictures and categories. categories can be manually edited in `notebooks/table_of_contents.js`. The title is taken from the first cell with an h1 in it, and the image is the last png (there's a GP notebook that is pure Bokeh!).
- Search UI is cleaner, and search logic is better (searching for "normal" now brings up the normal distribution first)
- Dedicated page to resources for learning about bayesian statistics with pymc3

image

image

### CODE REVIEW

**Comment by ColCarroll** [comment]
Oh, and note that since we only deploy docs on the latest release, we can either run this on the 3.5 tag and push manually, or wait till next version to actually deploy.

File: docs/source/learn.rst [hunk 1]
```diff
@@ -0,0 +1,252 @@
+.. title:: Learn
+
+.. raw:: html
+
+    <h1 class="ui header">Learn Bayesian statistics</h1>
+
+    <h2 class="ui header">...with others!</h2>
```
Comment (twiecki) [inline docs/source/learn.rst:7]: is this all hand-written?

Comment (ColCarroll) [reply docs/source/learn.rst:7]: yep. we _could_ do this from a configuration file or something, but I think copy/pasting for each new book is fine.

**Comment by twiecki** [comment]
yes definitely, just asking for my own understanding.

**Comment by twiecki** [comment]
Or we could cut a new release.

**Comment by twiecki** [comment]
I would definitely add this to the release-notes.

@ColCarroll Anything in specific you want feedback on? Otherwise we can probably just merge this.

**Comment by ColCarroll** [comment]
I'll add to readme, and also put the badges from github on the front page (forgot!)

I think in terms of feedback, mostly looking for "this is at least as good as current docs, and has some improvements". There are a bunch of rough edges that I would like to fix in smaller followup PRs (or at least open issues describing those so that anyone can tackle them).

**Comment by junpenglao** [comment]
It's a great improvement compare to our current doc, thanks for the hard work @ColCarroll!
Yeah we should cut a release after this.

**Comment by ColCarroll** [comment]
Should be all set. I also shuffled release notes a bit, since most folks are interpreting `sample_ppc` -> `sample_posterior_predictive` as a deprecation rather than maintenance.

**Pull request merged**